src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -352,93 +283,67 @@ def forward(
+            value_layer = torch.cat(
+                (past_value.type_as(value_layer), value_layer), dim=1
+            )  # [batch_size, k_length, num_heads, head_dim]
 
         if use_cache is True:
             present = (key_layer, value_layer)
         else:
             present = None
 
-        # [batch_size, head_dim, q_length, k_length]
-        output_size = (query_layer.size(0), query_layer.size(2), query_layer.size(1), key_layer.size(1))
-
-        # [batch_size, q_length, num_heads, head_dim] -> [q_length, batch_size * num_heads, head_dim]
-        query_layer = query_layer.transpose(1, 0).reshape(output_size[2], output_size[0] * output_size[1], -1)
-
-        # [batch_size, k_length, num_heads, head_dim] -> [k_length, batch_size * num_heads, head_dim]
-        key_layer = key_layer.transpose(1, 0).reshape(output_size[3], output_size[0] * output_size[1], -1)
-
-        # slice alibi tensor until the query length
-        sliced_alibi = alibi[: output_size[0] * output_size[1], :, : output_size[3]]
-
-        # Raw attention scores. [batch_size * num_heads, q_length, k_length]
         beta = 1.0 / self.layer_number
 
-        matmul_result = torch.baddbmm(
-            sliced_alibi,
-            query_layer.transpose(1, 0),
-            key_layer.transpose(1, 0).transpose(1, 2),
-            beta=beta,
-            alpha=(1.0 / self.norm_factor),
-        )
+        # [batch_size*num_heads, q_length, k_length]
+        matmul_result = (1.0 / self.norm_factor) * torch.bmm(
+            query_layer.transpose(1, 2).reshape(
+                -1, query_layer.shape[1], query_layer.shape[3]
+            ),  # [batch_size*num_heads, q_length, head_dim]
+            key_layer.permute(0, 2, 3, 1).reshape(
+                -1, key_layer.shape[3], key_layer.shape[1]
+            ),  # [batch_size*num_heads, head_dim, k_length]
+        ) + beta * alibi
 
         # change view to [batch_size, num_heads, q_length, k_length]
-        attention_scores = matmul_result.view(*output_size)
-
-        # attention scores and attention mask [b, np, sq, sk]
-        max_positions = max(attention_scores.shape[-1], attention_scores.shape[-2])
-        attention_probs = self.scale_mask_softmax(attention_scores, attention_mask, max_positions).to(
-            value_layer.dtype
+        attention_scores = matmul_result.view(-1, self.num_heads, matmul_result.size(1), matmul_result.size(2))
+
+        # We replace the scaled softmax by just a few line of code
+        input_dtype = attention_scores.dtype
+        attn_weights = (
+            attention_scores * self.layer_number
+        ) + attention_mask  # [batch_size, num_heads, q_length, k_length]
+        attn_weights = torch.max(attn_weights, torch.tensor(torch.finfo(attn_weights.dtype).min))
+        attention_probs = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(input_dtype) * (
+            ~attention_mask.bool()
         )
```

Comment:
Would be more readable if done on two lines.

Reply:
Should be fixed in adb67d64b90ce02cf016443f8afad80079214fff